packages/jaeger-ui/src/components/TracePage/TraceStatistics/index.tsx

```diff
@@ -392,79 +188,162 @@ export default class TraceStatistics extends Component<Props, State> {
     return allTableSpansChange;
   };
 
-  renderTableData() {
-    return this.state.tableValue.map(oneSpan => {
-      const {
-        count,
-        total,
-        avg,
-        min,
-        max,
-        selfTotal,
-        selfAvg,
-        selfMin,
-        selfMax,
-        percent,
-        color,
-        searchColor,
-        colorToPercent,
-      } = oneSpan;
-      const values: any[] = [count, total, avg, min, max, selfTotal, selfAvg, selfMin, selfMax, percent];
-      const uid = _.uniqueId('id');
-      if (!oneSpan.isDetail) {
-        return (
-          <MainTableData
-            key={uid}
-            type={oneSpan.type}
-            name={oneSpan.name}
-            searchColor={searchColor}
-            values={values}
-            columnsArray={columnsArray}
-            togglePopup={this.togglePopup}
-            valueNameSelector1={this.state.valueNameSelector1}
-            valueNameSelector2={this.state.valueNameSelector2}
-            color={color}
-            clickColumn={this.clickColumn}
-            colorToPercent={colorToPercent}
-          />
-        );
-      }
-      return (
-        <DetailTableData
-          key={uid}
-          type={oneSpan.type}
-          name={oneSpan.name}
-          searchColor={searchColor}
-          values={values}
-          columnsArray={columnsArray}
-          color={color}
-          togglePopup={this.togglePopup}
-          valueNameSelector2={this.state.valueNameSelector2}
-          colorToPercent={colorToPercent}
-        />
-      );
-    });
-  }
+  render() {
+    const onClickOption = (type: string, name: string) => {
+      if (this.state.valueNameSelector1 === 'sql.query' && type !== 'undefined') this.togglePopup(name);
+    };
 
-  renderTableHead() {
-    const { sortAsc, sortIndex } = this.state;
-    return (
-      <tr>
-        {columnsArray.map((element: any, index: number) => (
-          <HeaderTable
-            element={element}
-            key={element.title}
-            sortIndex={sortIndex}
-            index={index}
-            sortClick={this.sortClick}
-            sortAsc={sortAsc}
-          />
-        ))}
-      </tr>
-    );
-  }
+    const sorterFunction = <T extends keyof ITableSpan>(field: T): CompareFn<ITableSpan> => {
+      const sort = (a: ITableSpan, b: ITableSpan) => {
+        if (a.type === 'undefined') {
```

Comment:
Does `type` here have a similar meaning to `row.type`? Except for `Name` all other columns are numeric, so when is this `undefined`?

Reply:
Yes here also type has same meaning as row.type 
Here a.type is about each row not about sorting column.
Here a.type value is checking because
Basically we want to keep the undefined row at the bottom of subgroup irrespective of sorting so to attain this we are writing this conditions